### PR TITLE
feature/DGJ-601_delete-drafts

### PR DESCRIPTION
**Pull request by iman-jamali-fw**
## Summary
This PR allows drafts to be deleted as part of the  #601 ticket

## Changes
- An endpoint was added to webApi to delete drafts
- UIs and logic to delete drafts on the draft list page was added
- A confirmation modal was added along with success/failure toast message

## Screenshots (if applicable)

Screen Shot 2022-10-11 at 9 44 49 PM
Screen Shot 2022-10-11 at 9 44 36 PM

### CODE REVIEW

**Review by bhumin-fw — APPROVED** [review]
Looks good to me

**Review by simensma-fresh — APPROVED** [review]
Just one console log, otherwise looks great!

File: forms-flow-web/src/components/Draft/table.js [hunk 1]
```diff
@@ -119,7 +153,10 @@ export const columns = (lastModified, callback, t, redirectUrl) => {
     {
       dataField: "formUrl",
       text: <Translation>{(t) => t("Action")}</Translation>,
-      formatter: (cell, row) => linkDraft(cell, row, redirectUrl),
+      // formatter: (cell, row) => linkDraft(cell, row, redirectUrl),
```
[the remark this answers is not in the document]
Comment (iman-jamali-fw) [reply forms-flow-web/src/components/Draft/table.js:156]: I left this line to indicate a custom change for future upgrades.

File: forms-flow-web/src/components/Draft/List.js [hunk 2]
```diff
@@ -139,6 +150,43 @@ export const DraftList = React.memo(() => {
...
+                  draftName: "",
+                })
+              );
+            }}
+            onYes={() => {
+              dispatch(
+                deleteDraftById(
+                  selectedDraftForDelete.draftId,
+                  (error, data) => {
+                    console.log("#draft", { error, data });
```
[the remark this answers is not in the document]
Comment (iman-jamali-fw) [reply forms-flow-web/src/components/Draft/List.js:172]: Good catch!

**Pull request merged**